### ISSUE TITLE
/etc/sysconfig/rundeckd not being read after 3.0.1

### ISSUE THREAD

**Issue by sirvig** [rundeck/rundeck#3817]
**Describe the bug**
Using the RPM installation of Rundeck on RHEL7, after the 3.0.1 upgrade, the /etc/sysconfig/rundeckd file is not being read properly.   Per the directions in the /etc/rundeck/profile and online documentation, this is the file where overrides to /etc/rundeck/profile are supposed to live.   Subsequently my overrides for the ldap module were failing after the upgrade.

It looks like the init script dropped the line to read in the /etc/sysconfig/rundeckd file if it exists.

**My Rundeck detail**

* Rundeck version: 3.0.1.20180803-1.201808032143
* install type: rpm
* OS Name/version: RHEL 7.5
* DB Type/version: postgres

**To Reproduce**
cat /etc/sysconfig/rundeckd
```
RDECK_JVM_SETTINGS="${RDECK_JVM_SETTINGS:- -server -Drundeck.jaaslogin=true \
    -Djava.security.auth.login.config=/etc/rundeck/jaas-ldap.conf \
    -Dloginmodule.name=ldap}"

#export JAAS_CONF="${JAAS_CONF:-/etc/rundeck/jaas-ldap.conf}"
#export LOGIN_MODULE="${LOGIN_MODULE:-ldap}"
```

Relevant lines from /etc/init.d/rundeckd
```
# Source function library

prog="rundeckd"
RETVAL=0
PID_FILE=/var/run/${prog}.pid
servicelog=/var/log/rundeck/service.log

. /etc/rc.d/init.d/functions

. /etc/rundeck/profile
```

The default installation restart produces the following processes:
```
ps aux | grep rundeck
root     15164  0.0  0.0 181264  2176 ?        S    09:00   0:00 runuser -s /bin/bash -l rundeck -c java -Djava.security.auth.login.config=/etc/rundeck/jaas-loginmodule.conf            -Dloginmodule.name=RDpropertyfilelogin            -Drdeck.config=/etc/rundeck            -Drundeck.server.configDir=/etc/rundeck            -Dserver.datastore.path=/var/lib/rundeck/data/rundeck            -Drundeck.server.serverDir=/var/lib/rundeck            -Drdeck.projects=/var/lib/rundeck/projects            -Drdeck.runlogs=/var/lib/rundeck/logs            -Drundeck.config.location=/etc/rundeck/rundeck-config.properties            -Djava.io.tmpdir=/tmp/rundeck            -Drundeck.server.workDir=/tmp/rundeck            -Dserver.http.port=4440            -Drdeck.base=/var/lib/rundeck  -Xmx1024m -Xms256m -XX:MaxMetaspaceSize=256m -server  -jar /var/lib/rundeck/bootstrap/rundeck-3.0.1-20180803.war --skipinstall
rundeck  15167  193 11.7 3651408 718964 ?      Ssl  09:00   0:38 java -Djava.security.auth.login.config=/etc/rundeck/jaas-loginmodule.conf -Dloginmodule.name=RDpropertyfilelogin -Drdeck.config=/etc/rundeck -Drundeck.server.configDir=/etc/rundeck -Dserver.datastore.path=/var/lib/rundeck/data/rundeck -Drundeck.server.serverDir=/var/lib/rundeck -Drdeck.projects=/var/lib/rundeck/projects -Drdeck.runlogs=/var/lib/rundeck/logs -Drundeck.config.location=/etc/rundeck/rundeck-config.properties -Djava.io.tmpdir=/tmp/rundeck -Drundeck.server.workDir=/tmp/rundeck -Dserver.http.port=4440 -Drdeck.base=/var/lib/rundeck -Xmx1024m -Xms256m -XX:MaxMetaspaceSize=256m -server -jar /var/lib/rundeck/bootstrap/rundeck-3.0.1-20180803.war --skipinstall
```

Notice there is no mention of the ldap module configured in /etc/sysconfig/rundeckd

When I change the init script to be:
```
# Source function library

prog="rundeckd"
RETVAL=0
PID_FILE=/var/run/${prog}.pid
servicelog=/var/log/rundeck/service.log

. /etc/rc.d/init.d/functions

[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog   <<<<< added this line back in

. /etc/rundeck/profile
```

I get the correct setup for ldap module:
```
ps aux | grep rundeck
root     24143  0.0  0.0 181264  2172 ?        S    09:30   0:00 runuser -s /bin/bash -l rundeck -c java -Djava.security.auth.login.config=/etc/rundeck/jaas-loginmodule.conf            -Dloginmodule.name=RDpropertyfilelogin            -Drdeck.config=/etc/rundeck            -Drundeck.server.configDir=/etc/rundeck            -Dserver.datastore.path=/var/lib/rundeck/data/rundeck            -Drundeck.server.serverDir=/var/lib/rundeck            -Drdeck.projects=/var/lib/rundeck/projects            -Drdeck.runlogs=/var/lib/rundeck/logs            -Drundeck.config.location=/etc/rundeck/rundeck-config.properties            -Djava.io.tmpdir=/tmp/rundeck            -Drundeck.server.workDir=/tmp/rundeck            -Dserver.http.port=4440            -Drdeck.base=/var/lib/rundeck  -server -Drundeck.jaaslogin=true -Djava.security.auth.login.config=/etc/rundeck/jaas-ldap.conf -Dloginmodule.name=ldap  -jar /var/lib/rundeck/bootstrap/rundeck-3.0.1-20180803.war --skipinstall
rundeck  24146 10.8 15.4 4267956 947272 ?      Ssl  09:30   0:59 java -Djava.security.auth.login.config=/etc/rundeck/jaas-loginmodule.conf -Dloginmodule.name=RDpropertyfilelogin -Drdeck.config=/etc/rundeck -Drundeck.server.configDir=/etc/rundeck -Dserver.datastore.path=/var/lib/rundeck/data/rundeck -Drundeck.server.serverDir=/var/lib/rundeck -Drdeck.projects=/var/lib/rundeck/projects -Drdeck.runlogs=/var/lib/rundeck/logs -Drundeck.config.location=/etc/rundeck/rundeck-config.properties -Djava.io.tmpdir=/tmp/rundeck -Drundeck.server.workDir=/tmp/rundeck -Dserver.http.port=4440 -Drdeck.base=/var/lib/rundeck **-server -Drundeck.jaaslogin=true -Djava.security.auth.login.config=/etc/rundeck/jaas-ldap.conf -Dloginmodule.name=ldap** -jar /var/lib/rundeck/bootstrap/rundeck-3.0.1-20180803.war --skipinstall
```

**Expected behavior**
Upon restarting the service using "service rundeckd start/restart" or "systemctl start/restart rundeckd", the init script should read source both the /etc/rundeck/profile and /etc/sysconfig/rundeckd profile override variables to build the java execute command line.

**Additional context**
I am aware that I am using a settings change rather than overriding the variables.  It looks like the new version of grails requires java.security.auth.login.config as a passed parameter.

**Comment by gschueler** [comment]
hi, what is your /etc/rundeck/profile like? that was modified to source the /etc/sysconfig/rundeckd

**Comment by sirvig** [comment]
Sorry - more information here than you asked for.

tl;dr - I didnt run yum upgrade rundeck-config - only yum upgrade rundeck when going from 3.0.0 -> 3.0.1 (ultimately this caused this problem).  We probably need to update the documentation for upgrading.

My long exploration:
cat /etc/rundeck/profile
```
#########
# Rundeck Profile sourced from /etc/rc.d/init.d/rundeckd
#########
#
# NOTE: DO NOT MODIFY THIS FILE
# It will be replaced when the package is upgraded and your changes will not be saved.
#
# ##################
#
# To override variables in this file, you can instead create a file at:
#
#     /etc/sysconfig/rundeckd
#
# which contains exports for any of the variables listed below. E.g.:
#
#    export RUNDECK_TEMPDIR=/path/to/tmpdir
#
# That file will be sourced before this one, allowing your exports to take precedence.
#
###############

RDECK_INSTALL="${RDECK_INSTALL:-/var/lib/rundeck}"
RDECK_BASE="${RDECK_BASE:-/var/lib/rundeck}"
RDECK_CONFIG="${RDECK_CONFIG:-/etc/rundeck}"
RDECK_CONFIG_FILE="${RDECK_CONFIG_FILE:-$RDECK_CONFIG/rundeck-config.properties}"
RDECK_SERVER_BASE="${RDECK_SERVER_BASE:-$RDECK_BASE}"
RDECK_SERVER_CONFIG="${RDECK_SERVER_CONFIG:-$RDECK_CONFIG}"
RDECK_SERVER_DATA="${RDECK_SERVER_DATA:-$RDECK_BASE/data}"
RDECK_PROJECTS="${RDECK_PROJECTS:-$RDECK_BASE/projects}"
RUNDECK_TEMPDIR="${RUNDECK_TEMPDIR:-/tmp/rundeck}"
RUNDECK_WORKDIR="${RUNDECK_TEMPDIR:-$RDECK_BASE/work}"
RUNDECK_LOGDIR="${RUNDECK_LOGDIR:-$RDECK_BASE/logs}"
RDECK_JVM_SETTINGS="${RDECK_JVM_SETTINGS:- -Xmx1024m -Xms256m -XX:MaxMetaspaceSize=256m -server}"
RDECK_TRUSTSTORE_FILE="${RDECK_TRUSTSTORE_FILE:-$RDECK_CONFIG/ssl/truststore}"
RDECK_TRUSTSTORE_TYPE="${RDECK_TRUSTSTORE_TYPE:-jks}"
JAAS_CONF="${JAAS_CONF:-$RDECK_CONFIG/jaas-loginmodule.conf}"
LOGIN_MODULE="${LOGIN_MODULE:-RDpropertyfilelogin}"
RDECK_HTTP_PORT=${RDECK_HTTP_PORT:-4440}
RDECK_HTTPS_PORT=${RDECK_HTTPS_PORT:-4443}


# If no JAVA_CMD, try to find it in $JAVA_HOME
if [ -z "$JAVA_CMD" ] && [ -n "$JAVA_HOME" ] && [ -x "$JAVA_HOME/bin/java" ] ; then
  JAVA_CMD=$JAVA_HOME/bin/java
  PATH=$PATH:$JAVA_HOME/bin
  export JAVA_HOME
elif [ -z "$JAVA_CMD" ] ; then
  JAVA_CMD=java
fi

# build classpath without lone : that includes .
for jar in $(find $RDECK_INSTALL/cli -name '*.jar') ; do
  CLI_CP=${CLI_CP:+$CLI_CP:}$jar
done
for war in $(find $RDECK_INSTALL/bootstrap -name '*.war') ; do
  EXECUTABLE_WAR=$war
done

RDECK_JVM="-Djava.security.auth.login.config=$JAAS_CONF \
           -Dloginmodule.name=$LOGIN_MODULE \
           -Drdeck.config=$RDECK_CONFIG \
           -Drundeck.server.configDir=$RDECK_SERVER_CONFIG \
           -Dserver.datastore.path=$RDECK_SERVER_DATA/rundeck \
           -Drundeck.server.serverDir=$RDECK_INSTALL \
           -Drdeck.projects=$RDECK_PROJECTS \
           -Drdeck.runlogs=$RUNDECK_LOGDIR \
           -Drundeck.config.location=$RDECK_CONFIG_FILE \
           -Djava.io.tmpdir=$RUNDECK_TEMPDIR \
           -Drundeck.server.workDir=$RUNDECK_WORKDIR \
           -Dserver.http.port=$RDECK_HTTP_PORT \
           -Drdeck.base=$RDECK_BASE"
#
# Set min/max heap size
#
RDECK_JVM="$RDECK_JVM $RDECK_JVM_SETTINGS"
#
# SSL Configuration - Uncomment the following to enable.  Check SSL.properties for details.
#
if [ -n "$RUNDECK_WITH_SSL" ] ; then
  RDECK_SSL_OPTS="${RDECK_SSL_OPTS:- -Djavax.net.ssl.trustStore=$RDECK_TRUSTSTORE_FILE -Djavax.net.ssl.trustStoreType=$RDECK_TRUSTSTORE_TYPE -Djava.protocol.handler.pkgs=com.sun.net.ssl.internal.www.protocol}"
  RDECK_JVM="$RDECK_JVM -Drundeck.ssl.config=$RDECK_SERVER_CONFIG/ssl/ssl.properties -Dserver.https.port=${RDECK_HTTPS_PORT} ${RDECK_SSL_OPTS}"
fi

unset JRE_HOME

umask 002

rundeckd="$JAVA_CMD $RDECK_JVM $RDECK_JVM_OPTS -jar $EXECUTABLE_WAR --skipinstall"
```

FYI - I do not see an updated /etc/rundeck/profile being placed from the RPM file list:
yum list rundeck
```
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Installed Packages
rundeck.noarch                                         3.0.1.20180803-1.201808032143                                          @rundeck-release-bintray
```

repoquery -l rundeck.noarch
```
/etc/rc.d/init.d/rundeckd
/etc/rundeck
/tmp/rundeck
/usr/bin/rd-acl
/var/lib/rundeck
/var/lib/rundeck/.ssh
/var/lib/rundeck/bootstrap
/var/lib/rundeck/bootstrap/rundeck-3.0.1-20180803.war
/var/lib/rundeck/cli
/var/lib/rundeck/cli/ant-1.8.3.jar
/var/lib/rundeck/cli/ant-jsch-1.8.3.jar
/var/lib/rundeck/cli/ant-launcher-1.8.3.jar
/var/lib/rundeck/cli/commons-beanutils-1.9.3.jar
/var/lib/rundeck/cli/commons-cli-1.0.jar
/var/lib/rundeck/cli/commons-codec-1.10.jar
/var/lib/rundeck/cli/commons-collections-3.2.2.jar
/var/lib/rundeck/cli/commons-httpclient-3.0.1.jar
/var/lib/rundeck/cli/commons-lang-2.6.jar
/var/lib/rundeck/cli/commons-logging-1.2.jar
/var/lib/rundeck/cli/converter-jackson-2.2.0.jar
/var/lib/rundeck/cli/dom4j-2.1.0.jar
/var/lib/rundeck/cli/guava-24.1.1-jre.jar
/var/lib/rundeck/cli/jackson-annotations-2.8.0.jar
/var/lib/rundeck/cli/jackson-core-2.8.11.jar
/var/lib/rundeck/cli/jackson-databind-2.8.11.2.jar
/var/lib/rundeck/cli/jaxen-1.1.6.jar
/var/lib/rundeck/cli/jna-4.1.0.jar
/var/lib/rundeck/cli/jna-platform-4.1.0.jar
/var/lib/rundeck/cli/jsch-0.1.54.jar
/var/lib/rundeck/cli/jsch.agentproxy.connector-factory-0.0.9.jar
/var/lib/rundeck/cli/jsch.agentproxy.core-0.0.9.jar
/var/lib/rundeck/cli/jsch.agentproxy.jsch-0.0.9.jar
/var/lib/rundeck/cli/jsch.agentproxy.pageant-0.0.9.jar
/var/lib/rundeck/cli/jsch.agentproxy.sshagent-0.0.9.jar
/var/lib/rundeck/cli/jsch.agentproxy.usocket-jna-0.0.9.jar
/var/lib/rundeck/cli/jsch.agentproxy.usocket-nc-0.0.9.jar
/var/lib/rundeck/cli/log4j-1.2.17.jar
/var/lib/rundeck/cli/okhttp-3.6.0.jar
/var/lib/rundeck/cli/okio-1.11.0.jar
/var/lib/rundeck/cli/protobuf-java-3.5.1.jar
/var/lib/rundeck/cli/retrofit-2.2.0.jar
/var/lib/rundeck/cli/rundeck-core-3.0.1-20180803.jar
/var/lib/rundeck/cli/rundeck-storage-api-3.0.1-20180803.jar
/var/lib/rundeck/cli/rundeck-storage-conf-3.0.1-20180803.jar
/var/lib/rundeck/cli/rundeck-storage-data-3.0.1-20180803.jar
/var/lib/rundeck/cli/snakeyaml-1.14.jar
/var/lib/rundeck/data
/var/lib/rundeck/exp
/var/lib/rundeck/exp/webapp
/var/lib/rundeck/libext
/var/lib/rundeck/libext/cache
/var/lib/rundeck/libext/rundeck-copyfile-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-flow-control-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-git-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-jasypt-encryption-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-job-state-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-localexec-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-orchestrator-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-script-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-source-refresh-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-stub-plugin-3.0.1-20180803.jar
/var/lib/rundeck/libext/rundeck-upvar-plugin-3.0.1-20180803.jar
/var/lib/rundeck/logs
/var/lib/rundeck/var
/var/lib/rundeck/var/tmp
/var/lib/rundeck/var/tmp/pluginJars
/var/lib/rundeck/work
/var/log/rundeck
/var/rundeck/projects
```

Now that I am looking I see that the config files come from the rundeck-config rpm:
repoquery -f /etc/rundeck/profile
```
...
rundeck-config-0:3.0.1.20180803-1.201808032143.noarch
...
```

When I upgraded from rundeck 3.0.0 to rundeck 3.0.1, I only ran yum upgrade rundeck (I do not see anything in the upgrade documentation referencing a rundeck-config rpm.

Also none of the pre or postscripts reference the rundeck-config:
rpm -qp --scripts ./rundeck-3.0.1.20180803-1.201808032143.noarch.rpm
```
preinstall scriptlet (using /bin/sh):
getent group rundeck >/dev/null || groupadd -r rundeck
getent passwd rundeck >/dev/null || useradd -r -d /var/lib/rundeck -m -g rundeck rundeck
postinstall scriptlet (using /bin/sh):
if [ ! -e ~rundeck/.ssh/id_rsa ]; then
	su -c "ssh-keygen -q -t rsa -C '' -N '' -f ~rundeck/.ssh/id_rsa" rundeck
fi
/sbin/chkconfig --add rundeckd
preuninstall scriptlet (using /bin/sh):
if [ "$1" = 0 ]; then
    /sbin/service rundeckd stop >/dev/null 2>&1
	/sbin/chkconfig --del rundeckd
fi
postuninstall scriptlet (using /bin/sh):
if [ "$1" -ge "1" ] ; then
    /sbin/service rundeckd condrestart >/dev/null 2>&1 || :
fi
```

FYI - I will try this same upgrade on a clean machine.  The machine I am currently working with is running some testing jobs as we are evaluating Rundeck for production use.

**Comment by gschueler** [comment]
@sirvig thanks, I think you are right that the documentation needs to mention the rundeck-config rpm for upgrades.  Can we close this issue? I will file an issue on @rundeck/docs project to update the documentation

**Comment by gschueler** [comment]
fyi: i updated the download page for yum to mention `yum upgrade rundeck rundeck-config`
